src/apps/loan-list/materials/stackable-material/stackable-material.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useEffect, useCallback, FC, MouseEvent, useState } from "react";
+import { useText } from "../../../../core/utils/text";
+import {
+  FetchMaterial,
+  StackableMaterialProps,
+  MaterialProps
+} from "../utils/material-fetch-hoc";
+import { useModalButtonHandler } from "../../../../core/utils/modal";
+import DueDateLoansModal from "../../modal/due-date-loans-modal";
+import MaterialStatus from "./material-status";
+import MaterialOverdueLink from "./material-overdue-link";
+import AdditionalMaterialsButton from "./additional-materials-button";
+import MaterialInfo from "./material-info";
+
+const StackableMaterial: FC<StackableMaterialProps & MaterialProps> = ({
+  amountOfMaterialsWithDueDate,
+  material,
+  openModal,
+  selectMaterial,
+  loanMetaData,
+  dueDateLabel,
+  stack
+}) => {
+  const t = useText();
+  const { open } = useModalButtonHandler();
+  const [additionalMaterials] = useState(
+    amountOfMaterialsWithDueDate ? amountOfMaterialsWithDueDate - 1 : 0
+  );
+  const [showModal, setShowModal] = useState(false);
+  const { dueDate, id } = loanMetaData;
+
+  function stopPropagationFunction(e: Event | MouseEvent) {
```

Comment:
I think this should be global so we could use it everywhere

Reply:
can move